tck/mp-jwt-embedded/src/main/java/org/apache/tomee/microprofile/jwt/MPJWTInitializer.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *     Licensed to the Apache Software Foundation (ASF) under one or more
+ *     contributor license agreements.  See the NOTICE file distributed with
+ *     this work for additional information regarding copyright ownership.
+ *     The ASF licenses this file to You under the Apache License, Version 2.0
+ *     (the "License"); you may not use this file except in compliance with
+ *     the License.  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+package org.apache.tomee.microprofile.jwt;
+
+import org.eclipse.microprofile.auth.LoginConfig;
+
+import javax.servlet.FilterRegistration;
+import javax.servlet.ServletContainerInitializer;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.HandlesTypes;
+import javax.ws.rs.core.Application;
+import java.util.Set;
+
+/**
+ * Responsible for adding the filter into the chain and doing all other initialization
+ */
+@HandlesTypes(LoginConfig.class)
+public class MPJWTInitializer implements ServletContainerInitializer {
+
+    @Override
+    public void onStartup(final Set<Class<?>> classes, final ServletContext ctx) throws ServletException {
+
+        if (classes == null || classes.isEmpty()) {
+            return; // to REST application having @LoginConfig on it
+        }
+
+        for (Class<?> clazz : classes) {
+            final LoginConfig loginConfig = clazz.getAnnotation(LoginConfig.class);
+
+            if (loginConfig.authMethod() == null && !"MP-JWT".equals(loginConfig.authMethod())) {
+                continue;
+            }
+
+            if (!Application.class.isAssignableFrom(clazz)) {
+                continue; // do we really want Application?
```

Comment:
Does this need to be answered? Or are you happy with it?

Reply:
The MP-JWT spec requires an Application with the @LoginConfig annotation.
But the JAXRS spec does not require an Application. So we could for example have @LoginConfig straight into a REST Resource, wdyt?

Reply:
I can't find the mention in the spec that it is required on Application (only an example where they use it as such). WildFly Swarm also interpreted it as only the URL defined in the ApplicationPath will be protected.

Using it directly on the Rest resource installs the 'Global' (/*) MP JWT filter, also covering the other REST resources. Which maybe is confusing.

So probably opening an issue in the MP JWT for clarification is required.

Reply:
Good point Rudy. I'll create a ticket on MP-JWT